components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsFeatureTest.kt

```diff
@@ -913,6 +915,34 @@ class SitePermissionsFeatureTest {
         verify(transaction).remove(fragment)
     }
 
+    @Test
+    fun `getInitialSitePermissions - WHEN sitePermissionsRules is present the function MUST use the sitePermissionsRules values to create a SitePermissions object`() {
+
+        val rules = SitePermissionsRules(
+                location = SitePermissionsRules.Action.BLOCKED,
+                camera = SitePermissionsRules.Action.ASK_TO_ALLOW,
+                notification = SitePermissionsRules.Action.ASK_TO_ALLOW,
+                microphone = SitePermissionsRules.Action.BLOCKED,
+                autoplayAudible = SitePermissionsRules.Action.BLOCKED,
+                autoplayInaudible = SitePermissionsRules.Action.ALLOWED
+        )
+
+        sitePermissionFeature.sitePermissionsRules = rules
+
+        val mockPermissionRequest: PermissionRequest = mock()
+        whenever(mockPermissionRequest.uri).thenReturn(" http://mozilla.org")
```

Comment:
This is unimportant and 100% optional (I just think it's cool!), but I recently added a fun extension to `mock` that keeps all setup for one object in a self contained scope.

```kotlin
val mockPermissionRequest: PermissionRequest = mock {
    whenever(uri).thenReturn(" http://mozilla.org")
}
```

Reply:
Awesome ❤️  adding it :)